extensions/pkg/controller/worker/genericactuator/interface.go

```diff
@@ -41,6 +41,7 @@ type WorkerDelegate interface {
 	DeployMachineClasses(context.Context) error
 
 	// GenerateMachineDeployments generates the configuration for the desired machine deployments.
+	// Workerpool name should be substring of the machine deployment name.
```

Comment:
I do not think this is stable?. 
What happens if 
- workerName: worker-12
- workerName: worker-1

The machine deployment for `worker-12` would match both the worker pool `worker-1` and `worker-12`.
When looping over an unsorted machineDeployment slice this could lead to changes to the MachineDeployment specifications leading to an unwanted rolling update.

Reply:
Thanks for pointing, I was in the dilemma, but I just realized,  should have a worker-pool name as part of its labels, I'll give it a try.

Reply:
Yes, the machine deployment should have a label with the worker pool name I think.

Reply:
>The machine deployment for worker-12 would match both the worker pool worker-1 and worker-12.
When looping over an unsorted machineDeployment slice this could lead to changes to the MachineDeployment specifications leading to an unwanted rolling update.

Nice catch @danielfoehrKn . 

Yes, I think the label way would be preferred.

Reply:
Alternatively, you can let the extension controllers provide the to-be-used settings. The same is done for labels/taints/annotations, see https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/controller/worker/machines.go#L190-L192.
It has the advantage that you don't need any code here in the `genericactuator` (and no label matching or similar), but the "disadvantage" (not sure that it really is one) that the worker controllers have to be adapted. However, there it's easier because they are looping over the worker pools already.

Reply:
Sure, I am fine either way. Agree that if we want to be consistent, we can simply move the logic to the providers, and not touch the generic actuator. I'll remove the code changes from here then.

Reply:
I removed the changes from the genericactuator, and will follow-up on respected providers to adopt it. Ref: https://github.com/gardener/gardener/pull/2563/commits/ff6ab8d206cec92c0fba7529c268b33c85a853e7